lib/miq_expression.rb

```diff
@@ -909,7 +909,7 @@ def self.build_relats(model, parent = {}, seen = [])
     parent[:class_path] ||= model.name
     parent[:assoc_path] ||= model.name
     parent[:root] ||= model.name
-    result = {:columns => model.attribute_names, :parent => parent}
+    result = {:columns => model.public_attribute_names, :parent => parent}
```

Comment:
The `public_attribute_names` is a good addition here, and really cleaned this up.  👍